plugins/basic/nexus-repository-docker/src/main/java/org/sonatype/nexus/repository/docker/internal/DockerConnectorFacetImpl.java

```diff
@@ -0,0 +1,245 @@
+
+  @Inject
+  public DockerConnectorFacetImpl(final RepositoryManager repositoryManager,
+                                  final ManualValidator manualValidator,
+                                  final ConnectorRegistrar connectorRegistrar)
+  {
+    this.repositoryManager = checkNotNull(repositoryManager);
+    this.manualValidator = checkNotNull(manualValidator);
+    this.connectorRegistrar = checkNotNull(connectorRegistrar);
+  }
+
+  @Override
+  protected void doValidate(final Configuration configuration) throws Exception {
+    facet(ConfigurationFacet.class).validateSection(configuration, CONFIG_KEY, Config.class);
+    Config config = facet(ConfigurationFacet.class).readSection(configuration, CONFIG_KEY, Config.class);
+    Set<ConstraintViolation<?>> violations = new HashSet<>();
+    maybeAdd(violations, validatePort(pathOf("httpPort"), config.httpPort));
+    maybeAdd(violations, validatePort(pathOf("httpsPort"), config.httpsPort));
+    ConstraintViolations.maybePropagate(violations, log);
+  }
+
+  @Override
+  protected void doConfigure(final Configuration configuration) throws Exception {
+    config = facet(ConfigurationFacet.class).readSection(configuration, CONFIG_KEY, Config.class);
+    log.debug("Config: {}", config);
+  }
+
+  @Override
+  protected void doStart() throws Exception {
+    if (config.httpPort != null) {
+      try {
+        httpConfig = new DockerConnectorConfiguration(getRepository().getName(), HttpScheme.HTTP, config.httpPort);
+        connectorRegistrar.addConnector(httpConfig);
+      }
+      catch (Exception e) {
+        log.warn(
+            "Could not configure HTTP connector on port {} for docker repository {}",
+            config.httpPort, getRepository().getName(), e
+        );
+      }
+    }
+    if (config.httpsPort != null) {
+      try {
+        httpsConfig = new DockerConnectorConfiguration(getRepository().getName(), HttpScheme.HTTPS, config.httpsPort);
+        connectorRegistrar.addConnector(httpsConfig);
+      }
+      catch (Exception e) {
+        log.warn(
+            "Could not configure HTTPS connector on port {} for docker repository {}",
+            config.httpsPort, getRepository().getName(), e
+        );
+      }
+    }
+  }
+
+  @Override
+  protected void doStop() throws Exception {
+    if (httpConfig != null) {
+      connectorRegistrar.removeConnector(httpConfig);
+    }
+    if (httpsConfig != null) {
+      connectorRegistrar.removeConnector(httpsConfig);
+    }
+  }
+
+  @Override
+  public Integer getHttpPort() {
+    return config.httpPort;
+  }
+
+  @Override
+  public Integer getHttpsPort() {
+    return config.httpsPort;
+  }
+
+  private ConstraintViolation<?> validatePort(final String path, final Integer port) {
+    if (port != null) {
+      try {
+        for (Repository repository : repositoryManager.browse()) {
+          if (DockerFormat.NAME.equals(repository.getFormat().getValue()) && !repository.equals(getRepository())) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)